[2025-06-09 15:16:46,849: INFO: main: Welcome to our custom logging data science ]
[2025-06-10 16:55:08,658: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 16:56:22,568: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 16:56:52,037: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 16:56:52,040: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 16:56:52,045: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 16:57:45,914: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 16:57:45,934: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 16:57:45,952: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 16:57:57,047: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 16:57:57,047: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 16:57:57,055: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 16:59:22,579: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 16:59:22,579: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 16:59:22,591: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 17:02:37,255: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 17:02:37,256: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 17:02:37,263: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 17:08:55,215: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 17:08:55,223: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 17:08:55,226: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 17:08:55,234: INFO: common: created directory at:artifacts]
[2025-06-10 17:08:55,239: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-10 17:09:14,902: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 17:09:14,908: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 17:09:14,911: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 17:09:14,917: INFO: common: created directory at:artifacts]
[2025-06-10 17:09:14,923: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-10 17:10:11,447: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 17:10:11,454: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 17:10:11,460: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 17:10:11,462: INFO: common: created directory at:artifacts]
[2025-06-10 17:10:11,466: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-10 17:12:05,654: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 17:12:05,660: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 17:12:05,668: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 17:14:25,602: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 17:14:25,610: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 17:14:25,615: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 17:14:25,623: INFO: common: created directory at:artifacts]
[2025-06-10 17:14:25,628: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-10 17:17:40,697: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 17:17:40,710: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 17:17:40,719: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 17:17:40,725: INFO: common: created directory at:artifacts]
[2025-06-10 17:17:40,731: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-10 17:18:44,557: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 17:18:44,563: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 17:18:44,570: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 17:18:44,579: INFO: common: created directory at:artifacts]
[2025-06-10 17:18:44,585: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-10 17:18:56,177: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 17:18:56,180: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 17:18:56,180: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 17:18:56,191: INFO: common: created directory at:artifacts]
[2025-06-10 17:18:56,194: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-10 17:19:29,925: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 17:19:29,931: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 17:19:29,936: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 17:19:29,941: INFO: common: created directory at:artifacts]
[2025-06-10 17:19:29,947: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-10 17:21:21,017: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 17:21:21,017: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 17:21:21,028: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 17:21:21,034: INFO: common: created directory at:artifacts]
[2025-06-10 17:21:21,038: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-10 17:22:14,504: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 17:22:14,509: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 17:22:14,509: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 17:22:14,522: INFO: common: created directory at:artifacts]
[2025-06-10 17:22:14,526: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-10 17:22:24,999: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 17:22:25,005: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 17:22:25,011: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 17:22:25,019: INFO: common: created directory at:artifacts]
[2025-06-10 17:22:25,021: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-10 17:24:55,359: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 17:24:55,365: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 17:24:55,371: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 17:24:55,376: INFO: common: created directory at:artifacts]
[2025-06-10 17:24:55,381: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-10 17:25:52,117: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 17:25:52,127: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 17:25:52,131: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 17:25:52,138: INFO: common: created directory at:artifacts]
[2025-06-10 17:25:52,139: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-10 17:27:41,820: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 17:27:41,825: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 17:27:41,833: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 17:27:41,837: INFO: common: created directory at:artifacts]
[2025-06-10 17:27:41,844: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-10 17:28:21,660: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 17:28:21,668: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 17:28:21,673: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 17:28:21,673: INFO: common: created directory at:artifacts]
[2025-06-10 17:28:21,686: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-10 17:28:36,284: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 17:28:36,294: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 17:28:36,300: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 17:28:36,306: INFO: common: created directory at:artifacts]
[2025-06-10 17:28:36,312: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-10 17:30:47,092: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 17:30:47,097: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 17:30:47,101: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 17:30:47,104: INFO: common: created directory at:artifacts]
[2025-06-10 17:30:47,105: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-10 18:14:14,365: INFO: main: >>>>> stage Data Ingestion stage started <<<<<<]
[2025-06-10 18:14:14,365: ERROR: main: 'DataIngestionTrainingPipeline' object has no attribute 'main']
Traceback (most recent call last):
  File "C:\Users\layek\datascienceproject\main.py", line 9, in <module>
    obj.main()
AttributeError: 'DataIngestionTrainingPipeline' object has no attribute 'main'
[2025-06-10 18:16:49,143: INFO: main: >>>>> stage Data Ingestion stage started <<<<<<]
[2025-06-10 18:16:49,143: ERROR: main: 'DataIngestionTrainingPipeline' object has no attribute 'main']
Traceback (most recent call last):
  File "C:\Users\layek\datascienceproject\main.py", line 9, in <module>
    obj.main()
AttributeError: 'DataIngestionTrainingPipeline' object has no attribute 'main'
[2025-06-10 18:19:48,456: INFO: main: >>>>> stage Data Ingestion stage started <<<<<<]
[2025-06-10 18:19:48,456: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 18:19:48,456: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 18:19:48,456: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 18:19:48,468: INFO: common: created directory at:artifacts]
[2025-06-10 18:19:48,470: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-10 18:19:48,470: ERROR: main: 'DataIngestion' object has no attribute 'download_file']
Traceback (most recent call last):
  File "C:\Users\layek\datascienceproject\main.py", line 9, in <module>
    obj.initiate_data_ingestion()
  File "C:\Users\layek\datascienceproject\src\datascience\pipeline\data_ingestion_pipeline.py", line 16, in initiate_data_ingestion
    data_ingestion.download_file()
AttributeError: 'DataIngestion' object has no attribute 'download_file'
[2025-06-10 18:20:17,138: INFO: main: >>>>> stage Data Ingestion stage started <<<<<<]
[2025-06-10 18:20:17,141: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 18:20:17,143: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 18:20:17,144: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 18:20:17,145: INFO: common: created directory at:artifacts]
[2025-06-10 18:20:17,147: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-10 18:20:17,162: INFO: main: >>>>> stage Data Ingestion stage completed <<<

x=====x]
[2025-06-10 23:19:25,415: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2025-06-10 23:19:25,428: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 23:19:25,441: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 23:19:25,452: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 23:19:25,461: INFO: common: created directory at:artifacts]
[2025-06-10 23:19:25,463: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-10 23:19:25,465: ERROR: main: DataIngestionConfig.__init__() got an unexpected keyword argument 'source_URL']
Traceback (most recent call last):
  File "C:\Users\layek\datascienceproject\main.py", line 12, in <module>
    data_ingestion.initiate_data_ingestion()
  File "C:\Users\layek\datascienceproject\src\datascience\pipeline\data_ingestion_pipeline.py", line 14, in initiate_data_ingestion
    data_ingestion_config=config.get_data_ingestion_config()
  File "C:\Users\layek\datascienceproject\src\datascience\config\configuration.py", line 24, in get_data_ingestion_config
    data_ingestion_config=DataIngestionConfig(
TypeError: DataIngestionConfig.__init__() got an unexpected keyword argument 'source_URL'
[2025-06-10 23:20:35,468: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2025-06-10 23:20:35,468: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 23:20:35,477: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 23:20:35,486: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 23:20:35,489: INFO: common: created directory at:artifacts]
[2025-06-10 23:20:35,499: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-10 23:20:35,574: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2025-06-10 23:20:35,574: INFO: main: >>>>>> stage Data Validation stage started <<<<<<]
[2025-06-10 23:20:35,574: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 23:20:35,574: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 23:20:35,586: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 23:20:35,586: INFO: common: created directory at:artifacts]
[2025-06-10 23:20:35,586: ERROR: main: "'ConfigBox' object has no attribute 'data_validation'"]
Traceback (most recent call last):
  File "box\\box.py", line 594, in box.box.Box.__getitem__
KeyError: 'data_validation'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "box\\box.py", line 633, in box.box.Box.__getattr__
  File "box\\box.py", line 621, in box.box.Box.__getitem__
box.exceptions.BoxKeyError: "'data_validation'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "box\\box.py", line 635, in box.box.Box.__getattr__
AttributeError: 'ConfigBox' object has no attribute 'data_validation'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "box\\config_box.py", line 29, in box.config_box.ConfigBox.__getattr__
  File "box\\box.py", line 649, in box.box.Box.__getattr__
box.exceptions.BoxKeyError: "'ConfigBox' object has no attribute 'data_validation'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "box\\box.py", line 594, in box.box.Box.__getitem__
KeyError: 'data_validation'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "box\\box.py", line 633, in box.box.Box.__getattr__
  File "box\\box.py", line 621, in box.box.Box.__getitem__
box.exceptions.BoxKeyError: "'data_validation'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "box\\box.py", line 635, in box.box.Box.__getattr__
AttributeError: 'ConfigBox' object has no attribute 'data_validation'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\layek\datascienceproject\main.py", line 23, in <module>
    data_ingestion.initiate_data_validation()
  File "C:\Users\layek\datascienceproject\src\datascience\pipeline\data_validation_pipeline.py", line 13, in initiate_data_validation
    data_validation_config = config.get_data_validation_config()
  File "C:\Users\layek\datascienceproject\src\datascience\config\configuration.py", line 34, in get_data_validation_config
    config = self.config.data_validation
  File "box\\config_box.py", line 31, in box.config_box.ConfigBox.__getattr__
  File "box\\box.py", line 649, in box.box.Box.__getattr__
box.exceptions.BoxKeyError: "'ConfigBox' object has no attribute 'data_validation'"
[2025-06-10 23:27:47,519: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2025-06-10 23:27:47,521: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 23:27:47,521: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 23:27:47,521: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 23:27:47,532: INFO: common: created directory at:artifacts]
[2025-06-10 23:27:47,534: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-10 23:27:47,549: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2025-06-10 23:27:47,549: INFO: main: >>>>>> stage Data Validation stage started <<<<<<]
[2025-06-10 23:27:47,557: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 23:27:47,560: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 23:27:47,567: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 23:27:47,567: INFO: common: created directory at:artifacts]
[2025-06-10 23:27:47,572: INFO: common: created directory at:artifacts/data_validation]
[2025-06-10 23:27:47,619: INFO: main: >>>>>> stage Data Validation stage completed <<<<<<

x==========x]
[2025-06-10 23:32:20,592: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2025-06-10 23:32:20,592: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 23:32:20,592: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 23:32:20,608: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 23:32:20,613: INFO: common: created directory at:artifacts]
[2025-06-10 23:32:20,613: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-10 23:32:20,627: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2025-06-10 23:32:20,627: INFO: main: >>>>>> stage Data Validation stage started <<<<<<]
[2025-06-10 23:32:20,630: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 23:32:20,630: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 23:32:20,638: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 23:32:20,638: INFO: common: created directory at:artifacts]
[2025-06-10 23:32:20,641: INFO: common: created directory at:artifacts/data_validation]
[2025-06-10 23:32:20,675: INFO: main: >>>>>> stage Data Validation stage completed <<<<<<

x==========x]
[2025-06-10 23:34:50,868: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2025-06-10 23:34:50,868: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 23:34:50,883: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 23:34:50,887: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 23:34:50,892: INFO: common: created directory at:artifacts]
[2025-06-10 23:34:50,892: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-10 23:34:50,912: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2025-06-10 23:34:50,916: INFO: main: >>>>>> stage Data Validation stage started <<<<<<]
[2025-06-10 23:34:50,919: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-10 23:34:50,919: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-10 23:34:50,928: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-10 23:34:50,928: INFO: common: created directory at:artifacts]
[2025-06-10 23:34:50,928: INFO: common: created directory at:artifacts/data_validation]
[2025-06-10 23:34:50,928: INFO: data_validation: Reading file from: artifacts\data_ingestion\winequality-red.csv]
[2025-06-10 23:34:50,968: WARNING: data_validation: Column mismatch. Found: ['fixed acidity;"volatile acidity";"citric acid";"residual sugar";"chlorides";"free sulfur dioxide";"total sulfur dioxide";"density";"pH";"sulphates";"alcohol";"quality"'], Expected: ['fixed acidity', 'volatile acidity', 'citric acid', 'residual sugar', 'chlorides', 'free sulfur dioxide', 'total sulfur dioxide', 'density', 'pH', 'sulphates', 'alcohol', 'quality']]
[2025-06-10 23:34:50,968: INFO: main: >>>>>> stage Data Validation stage completed <<<<<<

x==========x]
[2025-06-11 00:07:36,428: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2025-06-11 00:07:36,456: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-11 00:07:36,461: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-11 00:07:36,467: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-11 00:07:36,467: INFO: common: created directory at:artifacts]
[2025-06-11 00:07:36,469: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-11 00:07:36,486: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2025-06-11 00:07:36,486: INFO: main: >>>>>> stage Data Validation stage started <<<<<<]
[2025-06-11 00:07:36,496: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-11 00:07:36,496: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-11 00:07:36,505: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-11 00:07:36,506: INFO: common: created directory at:artifacts]
[2025-06-11 00:07:36,509: INFO: common: created directory at:artifacts/data_validation]
[2025-06-11 00:07:36,546: INFO: main: >>>>>> stage Data Validation stage completed <<<<<<

x==========x]
[2025-06-11 00:07:36,546: INFO: main: >>>>>> stage Data Transformation stage started <<<<<<]
[2025-06-11 00:07:36,549: INFO: main: >>>>>> stage Data Transformation stage completed <<<<<<

x==========x]
[2025-06-11 00:18:22,602: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2025-06-11 00:18:22,606: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-11 00:18:22,616: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-11 00:18:22,624: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-11 00:18:22,624: INFO: common: created directory at:artifacts]
[2025-06-11 00:18:22,636: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-11 00:18:22,654: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2025-06-11 00:18:22,654: INFO: main: >>>>>> stage Data Validation stage started <<<<<<]
[2025-06-11 00:18:22,662: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-11 00:18:22,664: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-11 00:18:22,670: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-11 00:18:22,672: INFO: common: created directory at:artifacts]
[2025-06-11 00:18:22,674: INFO: common: created directory at:artifacts/data_validation]
[2025-06-11 00:18:22,716: INFO: main: >>>>>> stage Data Validation stage completed <<<<<<

x==========x]
[2025-06-11 00:18:22,716: INFO: main: >>>>>> stage Data Transformation stage started <<<<<<]
[2025-06-11 00:18:22,737: INFO: main: >>>>>> stage Data Transformation stage completed <<<<<<

x==========x]
[2025-06-11 00:20:24,684: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2025-06-11 00:20:24,693: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-11 00:20:24,696: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-11 00:20:24,696: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-11 00:20:24,696: INFO: common: created directory at:artifacts]
[2025-06-11 00:20:24,705: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-11 00:20:24,720: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2025-06-11 00:20:24,721: INFO: main: >>>>>> stage Data Validation stage started <<<<<<]
[2025-06-11 00:20:24,732: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-11 00:20:24,734: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-11 00:20:24,737: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-11 00:20:24,741: INFO: common: created directory at:artifacts]
[2025-06-11 00:20:24,742: INFO: common: created directory at:artifacts/data_validation]
[2025-06-11 00:20:24,778: INFO: main: >>>>>> stage Data Validation stage completed <<<<<<

x==========x]
[2025-06-11 00:20:24,778: INFO: main: >>>>>> stage Data Transformation stage started <<<<<<]
[2025-06-11 00:20:24,816: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-11 00:20:24,816: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-11 00:20:24,824: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-11 00:20:24,827: INFO: common: created directory at:artifacts]
[2025-06-11 00:20:24,828: INFO: common: created directory at:artifacts/data_transformation]
[2025-06-11 00:20:24,867: INFO: data_transformation: Splited data into training and test sets]
[2025-06-11 00:20:24,867: INFO: data_transformation: (1199, 1)]
[2025-06-11 00:20:24,867: INFO: data_transformation: (400, 1)]
[2025-06-11 00:20:24,870: INFO: main: >>>>>> stage Data Transformation stage completed <<<<<<

x==========x]
[2025-06-11 00:34:13,903: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2025-06-11 00:34:13,913: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-11 00:34:13,919: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-11 00:34:13,926: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-11 00:34:13,929: INFO: common: created directory at:artifacts]
[2025-06-11 00:34:13,929: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-11 00:34:13,946: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2025-06-11 00:34:13,946: INFO: main: >>>>>> stage Data Validation stage started <<<<<<]
[2025-06-11 00:34:13,950: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-11 00:34:13,957: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-11 00:34:13,962: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-11 00:34:13,964: INFO: common: created directory at:artifacts]
[2025-06-11 00:34:13,966: INFO: common: created directory at:artifacts/data_validation]
[2025-06-11 00:34:14,017: INFO: main: >>>>>> stage Data Validation stage completed <<<<<<

x==========x]
[2025-06-11 00:34:14,017: INFO: main: >>>>>> stage Data Transformation stage started <<<<<<]
[2025-06-11 00:34:14,029: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-11 00:34:14,032: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-11 00:34:14,032: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-11 00:34:14,039: INFO: common: created directory at:artifacts]
[2025-06-11 00:34:14,040: INFO: common: created directory at:artifacts/data_transformation]
[2025-06-11 00:34:14,074: INFO: data_transformation: Splited data into training and test sets]
[2025-06-11 00:34:14,074: INFO: data_transformation: (1199, 1)]
[2025-06-11 00:34:14,074: INFO: data_transformation: (400, 1)]
[2025-06-11 00:34:14,074: INFO: main: >>>>>> stage Data Transformation stage completed <<<<<<

x==========x]
[2025-06-11 00:34:14,074: INFO: main: >>>>>> stage Model Trainer stage started <<<<<<]
[2025-06-11 00:34:14,080: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-11 00:34:14,080: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-11 00:34:14,089: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-11 00:34:14,094: INFO: common: created directory at:artifacts]
[2025-06-11 00:34:14,095: ERROR: main: "'ConfigBox' object has no attribute 'elasticnet'"]
Traceback (most recent call last):
  File "box\\box.py", line 594, in box.box.Box.__getitem__
KeyError: 'ElasticNet'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "box\\box.py", line 633, in box.box.Box.__getattr__
  File "box\\box.py", line 621, in box.box.Box.__getitem__
box.exceptions.BoxKeyError: "'ElasticNet'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "box\\box.py", line 635, in box.box.Box.__getattr__
AttributeError: 'ConfigBox' object has no attribute 'ElasticNet'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "box\\config_box.py", line 29, in box.config_box.ConfigBox.__getattr__
  File "box\\box.py", line 649, in box.box.Box.__getattr__
box.exceptions.BoxKeyError: "'ConfigBox' object has no attribute 'ElasticNet'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "box\\box.py", line 594, in box.box.Box.__getitem__
KeyError: 'elasticnet'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "box\\box.py", line 633, in box.box.Box.__getattr__
  File "box\\box.py", line 621, in box.box.Box.__getitem__
box.exceptions.BoxKeyError: "'elasticnet'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "box\\box.py", line 635, in box.box.Box.__getattr__
AttributeError: 'ConfigBox' object has no attribute 'elasticnet'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\layek\datascienceproject\main.py", line 45, in <module>
    data_ingestion.initiate_model_training()
  File "C:\Users\layek\datascienceproject\src\datascience\pipeline\model_trainer_pipeline.py", line 14, in initiate_model_training
    model_trainer_config = config.get_model_trainer_config()
  File "C:\Users\layek\datascienceproject\src\datascience\config\configuration.py", line 59, in get_model_trainer_config
    params = self.params.ElasticNet
  File "box\\config_box.py", line 31, in box.config_box.ConfigBox.__getattr__
  File "box\\box.py", line 649, in box.box.Box.__getattr__
box.exceptions.BoxKeyError: "'ConfigBox' object has no attribute 'elasticnet'"
[2025-06-11 00:36:57,817: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2025-06-11 00:36:57,823: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-11 00:36:57,833: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-11 00:36:57,839: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-11 00:36:57,843: INFO: common: created directory at:artifacts]
[2025-06-11 00:36:57,845: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-11 00:36:57,860: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2025-06-11 00:36:57,860: INFO: main: >>>>>> stage Data Validation stage started <<<<<<]
[2025-06-11 00:36:57,863: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-11 00:36:57,871: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-11 00:36:57,871: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-11 00:36:57,871: INFO: common: created directory at:artifacts]
[2025-06-11 00:36:57,879: INFO: common: created directory at:artifacts/data_validation]
[2025-06-11 00:36:57,940: INFO: main: >>>>>> stage Data Validation stage completed <<<<<<

x==========x]
[2025-06-11 00:36:57,944: INFO: main: >>>>>> stage Data Transformation stage started <<<<<<]
[2025-06-11 00:36:57,948: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-11 00:36:57,956: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-11 00:36:57,960: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-11 00:36:57,964: INFO: common: created directory at:artifacts]
[2025-06-11 00:36:57,964: INFO: common: created directory at:artifacts/data_transformation]
[2025-06-11 00:36:57,982: INFO: data_transformation: Splited data into training and test sets]
[2025-06-11 00:36:57,987: INFO: data_transformation: (1199, 1)]
[2025-06-11 00:36:57,987: INFO: data_transformation: (400, 1)]
[2025-06-11 00:36:57,987: INFO: main: >>>>>> stage Data Transformation stage completed <<<<<<

x==========x]
[2025-06-11 00:36:57,987: INFO: main: >>>>>> stage Model Trainer stage started <<<<<<]
[2025-06-11 00:36:57,987: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-11 00:36:57,998: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-11 00:36:58,002: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-11 00:36:58,004: INFO: common: created directory at:artifacts]
[2025-06-11 00:36:58,006: INFO: common: created directory at:artifacts/model_trainer]
[2025-06-11 00:36:58,060: ERROR: main: "['quality'] not found in axis"]
Traceback (most recent call last):
  File "C:\Users\layek\datascienceproject\main.py", line 45, in <module>
    data_ingestion.initiate_model_training()
  File "C:\Users\layek\datascienceproject\src\datascience\pipeline\model_trainer_pipeline.py", line 16, in initiate_model_training
    model_trainer_config.train()
  File "C:\Users\layek\datascienceproject\src\datascience\components\model_trainer.py", line 19, in train
    train_x = train_data.drop([self.config.target_column], axis=1)
  File "C:\Users\layek\datascienceproject\venv\lib\site-packages\pandas\core\frame.py", line 5581, in drop
    return super().drop(
  File "C:\Users\layek\datascienceproject\venv\lib\site-packages\pandas\core\generic.py", line 4788, in drop
    obj = obj._drop_axis(labels, axis, level=level, errors=errors)
  File "C:\Users\layek\datascienceproject\venv\lib\site-packages\pandas\core\generic.py", line 4830, in _drop_axis
    new_axis = axis.drop(labels, errors=errors)
  File "C:\Users\layek\datascienceproject\venv\lib\site-packages\pandas\core\indexes\base.py", line 7070, in drop
    raise KeyError(f"{labels[mask].tolist()} not found in axis")
KeyError: "['quality'] not found in axis"
[2025-06-11 00:40:15,721: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2025-06-11 00:40:15,721: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-11 00:40:15,739: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-11 00:40:15,745: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-11 00:40:15,747: INFO: common: created directory at:artifacts]
[2025-06-11 00:40:15,747: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-11 00:40:15,768: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2025-06-11 00:40:15,770: INFO: main: >>>>>> stage Data Validation stage started <<<<<<]
[2025-06-11 00:40:15,774: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-11 00:40:15,779: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-11 00:40:15,782: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-11 00:40:15,787: INFO: common: created directory at:artifacts]
[2025-06-11 00:40:15,787: INFO: common: created directory at:artifacts/data_validation]
[2025-06-11 00:40:15,849: INFO: main: >>>>>> stage Data Validation stage completed <<<<<<

x==========x]
[2025-06-11 00:40:15,849: INFO: main: >>>>>> stage Data Transformation stage started <<<<<<]
[2025-06-11 00:40:15,857: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-11 00:40:15,861: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-11 00:40:15,867: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-11 00:40:15,870: INFO: common: created directory at:artifacts]
[2025-06-11 00:40:15,870: INFO: common: created directory at:artifacts/data_transformation]
[2025-06-11 00:40:15,927: INFO: data_transformation: Splited data into training and test sets]
[2025-06-11 00:40:15,928: INFO: data_transformation: (1199, 12)]
[2025-06-11 00:40:15,930: INFO: data_transformation: (400, 12)]
[2025-06-11 00:40:15,930: INFO: main: >>>>>> stage Data Transformation stage completed <<<<<<

x==========x]
[2025-06-11 00:40:15,930: INFO: main: >>>>>> stage Model Trainer stage started <<<<<<]
[2025-06-11 00:40:15,930: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-11 00:40:15,942: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-11 00:40:15,945: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-11 00:40:15,950: INFO: common: created directory at:artifacts]
[2025-06-11 00:40:15,950: INFO: common: created directory at:artifacts/model_trainer]
[2025-06-11 00:40:16,011: ERROR: main: "['quality'] not found in axis"]
Traceback (most recent call last):
  File "C:\Users\layek\datascienceproject\main.py", line 45, in <module>
    data_ingestion.initiate_model_training()
  File "C:\Users\layek\datascienceproject\src\datascience\pipeline\model_trainer_pipeline.py", line 16, in initiate_model_training
    model_trainer_config.train()
  File "C:\Users\layek\datascienceproject\src\datascience\components\model_trainer.py", line 19, in train
    train_x = train_data.drop([self.config.target_column], axis=1)
  File "C:\Users\layek\datascienceproject\venv\lib\site-packages\pandas\core\frame.py", line 5581, in drop
    return super().drop(
  File "C:\Users\layek\datascienceproject\venv\lib\site-packages\pandas\core\generic.py", line 4788, in drop
    obj = obj._drop_axis(labels, axis, level=level, errors=errors)
  File "C:\Users\layek\datascienceproject\venv\lib\site-packages\pandas\core\generic.py", line 4830, in _drop_axis
    new_axis = axis.drop(labels, errors=errors)
  File "C:\Users\layek\datascienceproject\venv\lib\site-packages\pandas\core\indexes\base.py", line 7070, in drop
    raise KeyError(f"{labels[mask].tolist()} not found in axis")
KeyError: "['quality'] not found in axis"
[2025-06-11 00:46:39,412: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2025-06-11 00:46:39,421: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-11 00:46:39,424: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-11 00:46:39,430: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-11 00:46:39,435: INFO: common: created directory at:artifacts]
[2025-06-11 00:46:39,436: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-11 00:46:39,448: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2025-06-11 00:46:39,449: INFO: main: >>>>>> stage Data Validation stage started <<<<<<]
[2025-06-11 00:46:39,450: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-11 00:46:39,460: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-11 00:46:39,464: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-11 00:46:39,469: INFO: common: created directory at:artifacts]
[2025-06-11 00:46:39,469: INFO: common: created directory at:artifacts/data_validation]
[2025-06-11 00:46:39,517: INFO: main: >>>>>> stage Data Validation stage completed <<<<<<

x==========x]
[2025-06-11 00:46:39,517: INFO: main: >>>>>> stage Data Transformation stage started <<<<<<]
[2025-06-11 00:46:39,528: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-11 00:46:39,531: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-11 00:46:39,533: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-11 00:46:39,538: INFO: common: created directory at:artifacts]
[2025-06-11 00:46:39,540: INFO: common: created directory at:artifacts/data_transformation]
[2025-06-11 00:46:39,582: INFO: data_transformation: Splited data into training and test sets]
[2025-06-11 00:46:39,582: INFO: data_transformation: (1199, 12)]
[2025-06-11 00:46:39,582: INFO: data_transformation: (400, 12)]
[2025-06-11 00:46:39,584: INFO: main: >>>>>> stage Data Transformation stage completed <<<<<<

x==========x]
[2025-06-11 00:46:39,584: INFO: main: >>>>>> stage Model Trainer stage started <<<<<<]
[2025-06-11 00:46:39,584: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-11 00:46:39,595: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-11 00:46:39,599: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-11 00:46:39,602: INFO: common: created directory at:artifacts]
[2025-06-11 00:46:39,603: INFO: common: created directory at:artifacts/model_trainer]
[2025-06-11 00:46:39,678: INFO: main: >>>>>> stage Model Trainer stage completed <<<<<<

x==========x]
[2025-06-11 00:46:39,678: INFO: main: >>>>>> stage Model evaluation stage started <<<<<<]
[2025-06-11 00:46:39,685: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-11 00:46:39,690: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-11 00:46:39,694: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-11 00:46:39,698: INFO: common: created directory at:artifacts]
[2025-06-11 00:46:39,699: INFO: common: created directory at:artifacts/model_evaluation]
[2025-06-11 00:46:42,081: INFO: common: json file saved at: artifacts\model_evaluation\metrics.json]
[2025-06-11 00:46:42,094: ERROR: main: name 'content' is not defined]
Traceback (most recent call last):
  File "C:\Users\layek\datascienceproject\main.py", line 55, in <module>
    data_ingestion.initiate_model_evaluation()
  File "C:\Users\layek\datascienceproject\src\datascience\pipeline\model_evaluation_pipeline.py", line 15, in initiate_model_evaluation
    model_evaluation.log_into_mlflow()
  File "C:\Users\layek\datascienceproject\src\datascience\components\model_evaluation.py", line 50, in log_into_mlflow
    save_json(path=Path(self.config.metric_file_name), data=scores)
  File "C:\Users\layek\datascienceproject\venv\lib\site-packages\ensure\main.py", line 835, in __call__
    return self.f(*args, **kwargs)
  File "C:\Users\layek\datascienceproject\src\datascience\utils\common.py", line 44, in save_json
    return ConfigBox(content)
NameError: name 'content' is not defined
[2025-06-11 00:50:19,654: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2025-06-11 00:50:19,654: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-11 00:50:19,666: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-11 00:50:19,675: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-11 00:50:19,675: INFO: common: created directory at:artifacts]
[2025-06-11 00:50:19,679: INFO: common: created directory at:artifacts/data_ingestion]
[2025-06-11 00:50:19,691: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2025-06-11 00:50:19,691: INFO: main: >>>>>> stage Data Validation stage started <<<<<<]
[2025-06-11 00:50:19,691: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-11 00:50:19,703: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-11 00:50:19,708: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-11 00:50:19,711: INFO: common: created directory at:artifacts]
[2025-06-11 00:50:19,711: INFO: common: created directory at:artifacts/data_validation]
[2025-06-11 00:50:19,752: INFO: main: >>>>>> stage Data Validation stage completed <<<<<<

x==========x]
[2025-06-11 00:50:19,752: INFO: main: >>>>>> stage Data Transformation stage started <<<<<<]
[2025-06-11 00:50:19,755: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-11 00:50:19,764: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-11 00:50:19,767: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-11 00:50:19,771: INFO: common: created directory at:artifacts]
[2025-06-11 00:50:19,774: INFO: common: created directory at:artifacts/data_transformation]
[2025-06-11 00:50:19,807: INFO: data_transformation: Splited data into training and test sets]
[2025-06-11 00:50:19,807: INFO: data_transformation: (1199, 12)]
[2025-06-11 00:50:19,807: INFO: data_transformation: (400, 12)]
[2025-06-11 00:50:19,807: INFO: main: >>>>>> stage Data Transformation stage completed <<<<<<

x==========x]
[2025-06-11 00:50:19,807: INFO: main: >>>>>> stage Model Trainer stage started <<<<<<]
[2025-06-11 00:50:19,807: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-11 00:50:19,824: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-11 00:50:19,829: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-11 00:50:19,831: INFO: common: created directory at:artifacts]
[2025-06-11 00:50:19,831: INFO: common: created directory at:artifacts/model_trainer]
[2025-06-11 00:50:19,892: INFO: main: >>>>>> stage Model Trainer stage completed <<<<<<

x==========x]
[2025-06-11 00:50:19,892: INFO: main: >>>>>> stage Model evaluation stage started <<<<<<]
[2025-06-11 00:50:19,899: INFO: common: yaml file:config\config.yml loaded successfully]
[2025-06-11 00:50:19,902: INFO: common: yaml file:params.yml loaded successfully]
[2025-06-11 00:50:19,907: INFO: common: yaml file:schema.yml loaded successfully]
[2025-06-11 00:50:19,909: INFO: common: created directory at:artifacts]
[2025-06-11 00:50:19,911: INFO: common: created directory at:artifacts/model_evaluation]
[2025-06-11 00:50:21,739: INFO: common: json file saved at: artifacts\model_evaluation\metrics.json]
[2025-06-11 00:50:38,154: INFO: main: >>>>>> stage Model evaluation stage completed <<<<<<

x==========x]
